src/QuadForm/Herm/Mass.jl

```diff
@@ -199,14 +197,24 @@ end
 #
 ################################################################################
 
+@doc Markdown.doc"""
+    local_factor(L::HermLat, p) -> fmpq
+
+Given a definite hermitian lattice $L$ and a bad prime $p$, return the local density
+of $L$ at $p$.
+"""
 function local_factor(L::HermLat, p)
+  @req isdefinite(L) "Lattice must be definite"
   S = base_ring(L)
   E = nf(S)
   K = base_field(E)
   q = norm(p)
   lp = prime_decomposition(S, p)
   ram = length(lp) == 1 && lp[1][2] == 2
-  if ram && isdyadic(p)
+  if ram && iseven(q)
+    if ismaximal(L,p)[1]
+      return _local_factor_maximal(L, p)
```

Comment:
Seems to be the only line that is not covered by the tests.

Reply:
The only new line yes, but more lines are not covered because the database and all the other examples I could find don't cover all the possible cases unfortunately... I added it in case it will be needed at some point.

Reply:
Ah, ok, makes sense.